rancher2/resource_rancher2_pod_security_admission_configuration_template.go

```diff
@@ -0,0 +1,162 @@
+		},
+	}
+}
+
+func resourceRancher2PodSecurityAdmissionConfigurationTemplateCreate(d *schema.ResourceData, meta interface{}) error {
+	podSecurityAdmissionConfigurationTemplate, err := expandPodSecurityAdmissionConfigurationTemplate(d)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] Creating PodSecurityAdmissionConfigurationTemplate %s", podSecurityAdmissionConfigurationTemplate.Name)
+
+	client, err := meta.(*Config).ManagementClient()
+	if err != nil {
+		return err
+	}
+
+	newPodSecurityAdmissionConfigurationTemplate, err := client.PodSecurityAdmissionConfigurationTemplate.Create(podSecurityAdmissionConfigurationTemplate)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(newPodSecurityAdmissionConfigurationTemplate.ID)
+
+	return resourceRancher2PodSecurityAdmissionConfigurationTemplateRead(d, meta)
+}
+
+func resourceRancher2PodSecurityAdmissionConfigurationTemplateRead(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[INFO] Refreshing PodSecurityAdmissionConfigurationTemplate with ID %s", d.Id())
+	client, err := meta.(*Config).ManagementClient()
+	if err != nil {
+		return err
+	}
+
+	return resource.Retry(d.Timeout(schema.TimeoutRead), func() *resource.RetryError {
+		psact, err := client.PodSecurityAdmissionConfigurationTemplate.ByID(d.Id())
+		if err != nil {
+			if IsNotFound(err) || IsForbidden(err) {
+				log.Printf("[INFO] PodSecurityAdmissionConfigurationTemplate with ID %s not found.", d.Id())
+				d.SetId("")
+				return nil
+			}
+			return resource.NonRetryableError(err)
+		}
+
+		if err = flattenPodSecurityAdmissionConfigurationTemplate(d, psact); err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		return nil
+	})
+}
+
+func resourceRancher2PodSecurityAdmissionConfigurationTemplateUpdate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[INFO] Updating PodSecurityAdmissionConfigurationTemplate with ID %s", d.Id())
+	client, err := meta.(*Config).ManagementClient()
+	if err != nil {
+		return err
+	}
+
+	psact, err := client.PodSecurityAdmissionConfigurationTemplate.ByID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	update, err := expandPodSecurityAdmissionConfigurationTemplate(d)
+	if err != nil {
+		return err
+	}
+
+	_, err = client.PodSecurityAdmissionConfigurationTemplate.Update(psact, update)
+	if err != nil {
+		return err
+	}
+
+	return resourceRancher2PodSecurityAdmissionConfigurationTemplateRead(d, meta)
+}
+
+func resourceRancher2PodSecurityAdmissionConfigurationTemplateDelete(d *schema.ResourceData, meta interface{}) error {
+	id := d.Id()
+	log.Printf("[INFO] Deleting PodSecurityAdmissionConfigurationTemplate with ID %s", id)
+	client, err := meta.(*Config).ManagementClient()
+	if err != nil {
+		return err
+	}
+
+	psact, err := client.PodSecurityAdmissionConfigurationTemplate.ByID(id)
+	if err != nil {
+		if IsNotFound(err) || IsForbidden(err) {
+			log.Printf("[INFO] PodSecurityAdmissionConfigurationTemplate with ID %s not found.", id)
```

Comment:
Will we get the "forbidden" error if the PSACT is not found? This log suggests that's the case, but that seems strange.

Reply:
TBH, I did not hit the forbidden error myself while testing, but it looks like a custom to check the notFound error along with the forbidden error in this repo.  To make it reasonable, my best guess is that there is a chance to hit the forbidden error if TF requests the resource while it is being removed on the server side.